src/dune

```diff
@@ -1,4 +1,23 @@
+(library
+ (name file)
+ (modules file)
+ (public_name tiff.file)
+ (libraries fmt cstruct optint))
+
+(library
+ (name endian)
+ (modules endian)
+ (public_name tiff.endian)
+ (libraries cstruct))
+
+(library
+ (name ifd)
```

Comment:
We should still be able to export everything under the `tiff` library (i.e. no need to add these other library stanzas to the dune file). If you need to re-export the interfaces you can add things like 

```ocaml
module Ifd = Ifd
```

to `tiff.ml`.